File: December - 04/4.TargetSubarrayFinder.py
N, K = map(int, input("Enter N and K: ").split())
arr = list(map(int, input(f"Enter {N} elements: ").split()))
start = 0
current_sum = 0
found = False
for end in range(N):
    current_sum += arr[end]
    while current_sum > K and start <= end:
        current_sum -= arr[start]
        start += 1
    if current_sum == K:
        print(start, end)
        found = True
        break
if not found:
    print(-1, -1)

File: December - 04/4 Target Subarray Finder.c
#include <stdio.h>

int main() {
    int n, k;

    printf("Enter number of elements and target sum: ");
    scanf("%d %d", &n, &k);

    int arr[n];

    printf("Enter the elements:\n");
    for (int i = 0; i < n; i++) {
        scanf("%d", &arr[i]);
    }

    int start = 0;
    int sum = 0;

    for (int end = 0; end < n; end++) {
        sum = sum + arr[end];

        while (sum > k) {
            sum = sum - arr[start];
            start++;
        }

        if (sum == k) {
            printf("%d %d", start, end);
            return 0;
        }
    }

    printf("-1 -1");
    return 0;
}
